### PR TITLE
Task 3 (Pool) by Tihomir Kehayov  

### PR DESCRIPTION
**Pull request by tkehayov**
(pool pattern) implementing Pool pattern task.

### CODE REVIEW

File: test/com/clouway/PoolObjectTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  public void happyPath() throws InstantiationException, IllegalAccessException {
+    Pool<House> poolObject = new PoolObject<House>(3);
+    House house = poolObject.acquire(House.class);
+    house.addName("Hora");
+
+    assertThat("Hora", is(house.getName()));
+  }
+
+  @Test
+  public void createMoreInstancesThanGiven() throws InstantiationException, IllegalAccessException {
```
Comment (mgenov) [inline test/com/clouway/PoolObjectTest.java]: poolLimitIsReached sounds better I think

File: test/com/clouway/PoolObjectTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    Pool<House> poolObject = new PoolObject<House>(2);
+    poolObject.acquire(House.class);
+    poolObject.acquire(House.class);
+    poolObject.acquire(House.class);
+    poolObject.acquire(House.class);
+  }
+
+  @Test
+  public void releaseOneObject() throws InstantiationException, IllegalAccessException {
```
Comment (mgenov) [inline test/com/clouway/PoolObjectTest.java]: releaseWhatWasAcquired

File: src/com/clouway/PoolObject.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  public PoolObject(int instances) {
+    this.instances = instances;
+  }
+
+  @Override
+  public T acquire(Class<T> t) {
+    instanceCounter++;
+    if (instanceCounter > instances) {
+      throw new ObjectInstanceOverflowException();
```
Comment (mgenov) [inline src/com/clouway/PoolObject.java]: Doesn't look like a good name. Think for a better name.

Comment (tkehayov) [inline src/com/clouway/PoolObject.java]: done

File: test/com/clouway/PoolObjectTest.java (hunk 2 above)
Comment (tkehayov) [inline test/com/clouway/PoolObjectTest.java]: done

File: test/com/clouway/PoolObjectTest.java (hunk 1 above)
Comment (tkehayov) [inline test/com/clouway/PoolObjectTest.java]: done

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**